packages/grid/data-grid/src/tests/cells.DataGrid.test.tsx

```diff
@@ -63,4 +63,66 @@ describe('<DataGrid /> - Cells', () => {
     );
     expect(getCell(0, 0)).to.have.class('foobar');
   });
+
+  describe('type `boolean`', () => {
+    it('able to render number from provided `renderCell`', () => {
+      render(
+        <div style={{ width: 300, height: 500 }}>
+          <DataGrid
+            autoHeight={isJSDOM}
+            columns={[
+              {
+                field: 'number',
+                headerName: 'Number',
+                type: 'boolean',
+                renderCell: (params) => (params.value ? 1 : 0),
+              },
+            ]}
+            rows={[
+              {
+                id: 1,
+                number: false,
+              },
+              {
+                id: 2,
+                number: true,
+              },
+            ]}
+          />
+        </div>,
+      );
+      expect(getCell(0, 0)).to.have.text('0');
+      expect(getCell(1, 0)).to.have.text('1');
+    });
+
+    it('formattedValue to string by default', () => {
```

Comment:
I think this test could be moved to this file: 

https://github.com/mui-org/material-ui-x/blob/363a7a1cff3f4cb1dba20bb17264acf7d80d89f6/packages/grid/x-grid/src/tests/export.XGrid.test.tsx#L94

There, the value formatter is used by default and we don't need to provide a `renderCell`.

Reply:
Oh, but the fix is basic feature, not just for x-grid. should I move it?

Reply:
Yeah, I think we can move it. This file is for tests that use the imperative API, but, since this fix is for both DataGrid and XGrid and we don't have another file to use it, this one is fine.